include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+    BOOST_URL_CXX20_CONSTEXPR
+    std::basic_string<char, std::char_traits<char>, A>&
+    to_string(std::basic_string<char, std::char_traits<char>, A> & str) const
+    {
+        str.resize(size());
+        copy(&str[0], size());
+        return str;
+    }
+
+    /** Returns a basic_string
+
+        This overload constructs a new string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A = std::allocator<char>>
+    BOOST_URL_CXX20_CONSTEXPR
+    std::basic_string<char, std::char_traits<char>, A>
+    to_string() const
+    {
+        std::basic_string<
+            char, std::char_traits<char>, A> str(size(), ' ');
+        return to_string(str);
+    }
+
+    /** Returns a basic_string
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A>
+    BOOST_URL_CXX20_CONSTEXPR explicit
+    operator std::basic_string<char, std::char_traits<char>, A>() const
+    {
+        return to_string<A>();
+    }
+
+    /** Appends the decoded characters to a container
+
+        This overload allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class T>
+    BOOST_URL_CXX20_CONSTEXPR
+    T&
+    append_to(T& str) const;
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if two strings are equal after percent decoding
+      */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /// @copydoc operator==
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /// @copydoc operator==
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) == 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if two strings are not equal after percent decoding
+      */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /// @copydoc operator!=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /// @copydoc operator!=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) != 0;
+    }
+
+    /** Compares two strings
```

Comment:
```
/** Return true if lhs is less than rhs

    The comparison is performed lexicographically after
    percent-decoding is applied.
*/
```

Reply:
lhs and rhs don't look great, maybe `s0` and `s1`? What does cppreference use?

Reply:
cppreference uses `lhs` and `rhs`: https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp

should I still change that?

Reply:
How does it render? Can you paste a screen snippet?

Reply:
On second thought, yes I like `s0` and `s1` much better. Rationale: `s` is universally recognized as "string" and zero is less than one which reinforces the left and right sides in a more visual way. lhs and rhs have legibility problems.